### PR TITLE
Added FITS reading

### LINKED ISSUE #4054: Reading FITS images
Whenever I try to run PIL.FitsStubImagePlugin.register_handler()
it gives me an error saying PIL doesn't have any FitsStubImagePlugin module.
How do I open my fits file as it says I need a handler to operate .fits file

### PR DESCRIPTION
**Pull request by radarhere**
Resolves #4054, adding reading of FITS files by upgrading the FITS plugin from a StubImageFile to an ImageFile.

I have updated the hopper FITS test image. The original was similar, but not equal, to `hopper("L")`, so I generated this new file with Pillow and ImageMagick. The updated file is still L mode, like the original.

### CODE REVIEW

File: src/PIL/FitsImagePlugin.py [hunk 1]
```diff
@@ -75,26 +59,13 @@ def _open(self):
...
-
-    def _load(self):
-        return _handler
-
-
-def _save(im, fp, filename):
-    if _handler is None or not hasattr("_handler", "save"):
-        raise OSError("FITS save handler not installed")
-    _handler.save(im, fp, filename)
+        offset = math.ceil(self.fp.tell() / 2880) * 2880
```
Comment (radarhere) [inline src/PIL/FitsImagePlugin.py:62]: The documentation mentioned that the offset for the image data was a multiple of 2880.

https://fits.gsfc.nasa.gov/fits_primer.html
> Each header or data unit is a multiple of 2880 bytes long. If necessary, the header or data unit is padded out to the required length with ASCII blanks or NULLs depending on the type of unit.

> The data unit, if present, immediately follows the last 2880-byte block in the header unit.

File: docs/handbook/image-file-formats.rst [hunk 2]
```diff
@@ -1064,6 +1064,11 @@ is commonly used in fax applications. The DCX decoder can read files containing
 When the file is opened, only the first image is read. You can use
 :py:meth:`~PIL.Image.Image.seek` or :py:mod:`~PIL.ImageSequence` to read other images.
 
+FITS
+^^^^
+
```
Comment (hugovk) [inline docs/handbook/image-file-formats.rst:1069]:
```suggestion

.. versionadded:: 9.1.0

```

**Review by hugovk — APPROVED** [review]
Let's also add to release notes.

Do we need to worry about the removed `FITSStubImageFile` class?

File: docs/deprecations.rst [hunk 3]
```diff
@@ -133,6 +133,15 @@ Deprecated                                             Use instead
...
 =====================================================  ============================================================
 
+FitsStubImagePlugin
+~~~~~~~~~~~~~~~~~~~
+
+.. deprecated:: 9.1.0
+
+The stub image plugin FitsStubImagePlugin has been deprecated and will be removed in
+Pillow 10.0.0 (2023-07-01). FITS images can be read without a handler through
+FitsImagePlugin instead.
```
Comment (hugovk) [inline docs/deprecations.rst:143]:
```suggestion
The stub image plugin :mod:`~PIL.FitsStubImagePlugin` has been deprecated and will be removed in
Pillow 10.0.0 (2023-07-01). FITS images can be read without a handler through
:mod:`~PIL.FitsImagePlugin` instead.
```

File: docs/releasenotes/9.1.0.rst [hunk 4]
```diff
@@ -97,6 +97,15 @@ In effect, ``viewer.show_file("test.jpg")`` will continue to work unchanged.
...
 ``viewer.show_file(path="test.jpg")`` instead.
 
+FitsStubImagePlugin
+~~~~~~~~~~~~~~~~~~~
+
+.. deprecated:: 9.1.0
+
+The stub image plugin FitsStubImagePlugin has been deprecated and will be removed in
+Pillow 10.0.0 (2023-07-01). FITS images can be read without a handler through
+FitsImagePlugin instead.
```
Comment (hugovk) [inline docs/releasenotes/9.1.0.rst:107]:
```suggestion
The stub image plugin :mod:`~PIL.FitsStubImagePlugin` has been deprecated and will be removed in
Pillow 10.0.0 (2023-07-01). FITS images can be read without a handler through
:mod:`~PIL.FitsImagePlugin` instead.
```

**Review by hugovk — COMMENTED** [review]
Looks good, couple of little docs autolink nits.

**Comment by radarhere** [comment]
The links to FitsImagePlugin are now there - but I discovered I couldn't link to FitsStubImagePlugin, because I removed it from the documentation to prevent new users from finding it.

File: docs/releasenotes/9.1.0.rst [hunk 5]
```diff
@@ -97,6 +97,15 @@ In effect, ``viewer.show_file("test.jpg")`` will continue to work unchanged.
 ``viewer.show_file(file="test.jpg")`` will raise a deprecation warning, and suggest
 ``viewer.show_file(path="test.jpg")`` instead.
 
+FitsStubImagePlugin
+~~~~~~~~~~~~~~~~~~~
+
+.. deprecated:: 9.1.0
+
+The stub image plugin FitsStubImagePlugin has been deprecated and will be removed in
```
Comment (hugovk) [inline docs/releasenotes/9.1.0.rst:105]:
```suggestion
The stub image plugin ``FitsStubImagePlugin`` has been deprecated and will be removed in
```

File: docs/deprecations.rst [hunk 6]
```diff
@@ -133,6 +133,15 @@ Deprecated                                             Use instead
 ``PngImagePlugin.APNG_BLEND_OP_OVER``                  ``PngImagePlugin.Blend.OP_OVER``
 =====================================================  ============================================================
 
+FitsStubImagePlugin
+~~~~~~~~~~~~~~~~~~~
+
+.. deprecated:: 9.1.0
+
+The stub image plugin FitsStubImagePlugin has been deprecated and will be removed in
```
Comment (hugovk) [inline docs/deprecations.rst:141]:
```suggestion
The stub image plugin ``FitsStubImagePlugin`` has been deprecated and will be removed in
```

**Review by hugovk — COMMENTED** [review]
Makes sense, let's just code format it then.

**Review by hugovk — APPROVED** [review]

**Pull request merged**